### PR TITLE
Feature/11 implement entities

### PR DESCRIPTION
**Pull request by AyTee001**
https://trello.com/c/fN7CkCLf/11-implement-entities

### CODE REVIEW

File: backend/LeetWars.Core/LeetWars.Core.DAL/Context/ModelBuilderExtensions.cs [hunk 1]
```diff
@@ -1,33 +1,271 @@
...
+            new ChallengeLevel("Extreme"){ Id = 4, Reward = 40 }
+        };
+
+        private static readonly ICollection<Language> _languages = new List<Language>()
+        {
+            new Language("C#"){Id = 1},
+            new Language("Typescript"){Id = 2},
+            new Language("Python"){Id = 3},
+            new Language("Javascript"){Id = 4}
+        };
```
Comment (tatianahutii) [inline backend/LeetWars.Core/LeetWars.Core.DAL/Context/ModelBuilderExtensions.cs:56]: Move to a separate class\file

**Review by tatianahutii — CHANGES_REQUESTED** [review]
And please fix all code smells

File: backend/LeetWars.Core/LeetWars.Core.DAL/Context/ModelBuilderExtensions.cs [hunk 2]
```diff
@@ -1,33 +1,271 @@
...
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(ChallengeConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(ChallengeLevelConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(ChallengeVersionConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(LanguageConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(LanguageVersionConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(SubscriptionConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(SubscriptionTypeConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(TagConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(UserConfig).Assembly);
+            modelBuilder.ApplyConfigurationsFromAssembly(typeof(UserLanguageLevelConfig).Assembly);
```
[the remark this answers is not in the document]
Comment (AyTee001) [reply backend/LeetWars.Core/LeetWars.Core.DAL/Context/ModelBuilderExtensions.cs]: I modified the file and left just one statement

File: backend/LeetWars.Core/LeetWars.Core.DAL/Context/LeetWarsCoreContextFactory.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Design;
+using Microsoft.Extensions.Configuration;
+using System.IO;
+
+namespace LeetWars.Core.DAL.Context
+{
+    public class LeetWarsCoreContextFactory : IDesignTimeDbContextFactory<LeetWarsCoreContext>
```
[the remark this answers is not in the document]
Comment (AyTee001) [reply backend/LeetWars.Core/LeetWars.Core.DAL/Context/LeetWarsCoreContextFactory.cs]: This factory was utilized by me to work with ef core and db from terminal. I deleted this file

**Review by tatianahutii — APPROVED** [review]

**Pull request merged**